client/src/pages/Home.js

```diff
@@ -72,12 +72,41 @@ const Home = () => {
 	const handleChange = (setter) => (event) => {
 		setter(event.target.value);
 	};
+
 	const onOutputValueChange = handleChange(setOutputValue);
 	const onAlternativeValueChange = handleChange(setAlternativeValue);
 
-	const handleSubmit = (data) => {
-		setOutputValue(data);
-		setShowOutput(true);
+	const handleSubmit = () => {
+		if (inputValue.trim() !== "") {
+			setIsLoading(true); // Set loading state to true
+			fetchData();
+		}
+	};
+
```

Comment:
I think we can keep the original handleSubmit logic, and we don't need to set loading in here